golang/trace/zipkin_tracereporter.go

```diff
@@ -0,0 +1,100 @@
+package trace
+
+import (
+	"encoding/base64"
+	"encoding/binary"
+	"net"
+	"time"
+
+	tc "github.com/uber/tchannel/golang"
+	"github.com/uber/tchannel/golang/thrift"
+	"github.com/uber/tchannel/golang/trace/thrift/gen-go/tcollector"
+)
+
+// ZipkinTraceReporter is a trace reporter that submits trace spans in to zipkin trace server.
+type ZipkinTraceReporter struct {
+	tchannel *tc.Channel
+}
+
+// Report method will submit trace span to tcollector server.
+func (r *ZipkinTraceReporter) Report(
+	span tc.Span, annotations []tc.Annotation, binaryAnnotations []tc.BinaryAnnotation, name string, endpoint *tc.Endpoint) {
+
+	thriftClient := thrift.NewClient(r.tchannel, "tcollector", nil)
```

Comment:
You don't want to create a NewClient for every single Report. You should create one on creation and use the same client throughout.